pkg/sql/stats/forecast.go

```diff
@@ -44,11 +44,58 @@ var UseStatisticsForecasts = settings.RegisterBoolSetting(
 // required to produce a statistics forecast. Forecasts based on 1 or 2
 // observations will always have R² = 1 (perfect goodness of fit) regardless of
 // the accuracy of the forecast.
-const minObservationsForForecast = 3
+var minObservationsForForecast = settings.RegisterIntSetting(
+	settings.TenantWritable,
+	"sql.stats.forecasts.min_observations",
+	"the mimimum number of observed statistics required to produce a statistics forecast",
+	3,
+	func(v int64) error {
+		if v < 1 || v > math.MaxInt {
+			return errors.Errorf("expected value in range [%d, %d], got %d", 1, math.MaxInt, v)
+		}
+		return nil
+	},
+).WithPublic()
 
 // minGoodnessOfFit is the minimum R² (goodness of fit) measurement all
 // predictive models in a forecast must have for us to use the forecast.
-const minGoodnessOfFit = 0.95
+var minGoodnessOfFit = settings.RegisterFloatSetting(
+	settings.TenantWritable,
+	"sql.stats.forecasts.min_goodness_of_fit",
+	"the minimum R² (goodness of fit) measurement required from all predictive models to use a "+
+		"forecast",
+	0.95,
+	func(v float64) error {
+		if v < 0 || v > 1 {
+			return errors.Errorf("expected value in range [%f, %f], got %f", 0.0, 1.0, v)
+		}
+		return nil
+	},
+).WithPublic()
+
+// maxDecrease is the minimum ratio of a prediction to the lowest prior
+// observation that we allow. Predictions falling below this will be clamped to
+// the lower bound calculated from this ratio. This lower bound is needed to
+// prevent forecasting zero rows for downward-trending statistics, which can
+// cause bad plans when the forecast is initially used.
+//
+// This happens to be the same as unknownFilterSelectivity, but there's not a
+// strong theoretical reason for it.
+var maxDecrease = settings.RegisterFloatSetting(
+	settings.TenantWritable,
+	"sql.stats.forecasts.max_decrease",
+	"the most a prediction is allowed to decrease, expressed as the minimum ratio of the prediction "+
+		"to the lowest prior observation",
+	1.0/3.0,
```

Comment:
Should we set this to 0 be default in backports? Would that reduce risk of this change?

Reply:
Good point, I will do that.